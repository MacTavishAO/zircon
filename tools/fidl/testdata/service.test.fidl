// Copyright 2019 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

library test.name;

protocol FirstProtocol {
    MethodOnFirst();
};

service SingleMemberService {
    FirstProtocol single_member;
};

protocol SecondProtocol {
    MethodOnSecond();
};

service MultiHeterogeneousMemberService {
    FirstProtocol first_member;
    SecondProtocol second_member;
};

service MultiHomogeneousMemberService {
    FirstProtocol first_member;
    FirstProtocol second_member;
};
