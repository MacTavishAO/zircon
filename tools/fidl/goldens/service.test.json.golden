{
  "version": "0.0.1",
  "name": "test.name",
  "library_dependencies": [],
  "bits_declarations": [],
  "const_declarations": [],
  "enum_declarations": [],
  "experimental_resource_declarations": [],
  "interface_declarations": [
    {
      "name": "test.name/FirstProtocol",
      "location": {
        "filename": "service.test.fidl",
        "line": 7,
        "column": 10,
        "length": 13
      },
      "methods": [
        {
          "ordinal": 6344653839727902267,
          "name": "MethodOnFirst",
          "location": {
            "filename": "service.test.fidl",
            "line": 8,
            "column": 5,
            "length": 13
          },
          "has_request": true,
          "maybe_request": [],
          "maybe_request_type_shape_v1": {
            "inline_size": 16,
            "alignment": 8,
            "depth": 0,
            "max_handles": 0,
            "max_out_of_line": 0,
            "has_padding": false,
            "has_flexible_envelope": false,
            "is_resource": false
          },
          "has_response": false,
          "is_composed": false
        }
      ]
    },
    {
      "name": "test.name/SecondProtocol",
      "location": {
        "filename": "service.test.fidl",
        "line": 15,
        "column": 10,
        "length": 14
      },
      "methods": [
        {
          "ordinal": 6306721464597794337,
          "name": "MethodOnSecond",
          "location": {
            "filename": "service.test.fidl",
            "line": 16,
            "column": 5,
            "length": 14
          },
          "has_request": true,
          "maybe_request": [],
          "maybe_request_type_shape_v1": {
            "inline_size": 16,
            "alignment": 8,
            "depth": 0,
            "max_handles": 0,
            "max_out_of_line": 0,
            "has_padding": false,
            "has_flexible_envelope": false,
            "is_resource": false
          },
          "has_response": false,
          "is_composed": false
        }
      ]
    }
  ],
  "service_declarations": [
    {
      "name": "test.name/SingleMemberService",
      "location": {
        "filename": "service.test.fidl",
        "line": 11,
        "column": 9,
        "length": 19
      },
      "members": [
        {
          "type": {
            "kind": "identifier",
            "identifier": "test.name/FirstProtocol",
            "nullable": false
          },
          "name": "single_member",
          "location": {
            "filename": "service.test.fidl",
            "line": 12,
            "column": 19,
            "length": 13
          }
        }
      ]
    },
    {
      "name": "test.name/MultiHeterogeneousMemberService",
      "location": {
        "filename": "service.test.fidl",
        "line": 19,
        "column": 9,
        "length": 31
      },
      "members": [
        {
          "type": {
            "kind": "identifier",
            "identifier": "test.name/FirstProtocol",
            "nullable": false
          },
          "name": "first_member",
          "location": {
            "filename": "service.test.fidl",
            "line": 20,
            "column": 19,
            "length": 12
          }
        },
        {
          "type": {
            "kind": "identifier",
            "identifier": "test.name/SecondProtocol",
            "nullable": false
          },
          "name": "second_member",
          "location": {
            "filename": "service.test.fidl",
            "line": 21,
            "column": 20,
            "length": 13
          }
        }
      ]
    },
    {
      "name": "test.name/MultiHomogeneousMemberService",
      "location": {
        "filename": "service.test.fidl",
        "line": 24,
        "column": 9,
        "length": 29
      },
      "members": [
        {
          "type": {
            "kind": "identifier",
            "identifier": "test.name/FirstProtocol",
            "nullable": false
          },
          "name": "first_member",
          "location": {
            "filename": "service.test.fidl",
            "line": 25,
            "column": 19,
            "length": 12
          }
        },
        {
          "type": {
            "kind": "identifier",
            "identifier": "test.name/FirstProtocol",
            "nullable": false
          },
          "name": "second_member",
          "location": {
            "filename": "service.test.fidl",
            "line": 26,
            "column": 19,
            "length": 13
          }
        }
      ]
    }
  ],
  "struct_declarations": [],
  "table_declarations": [],
  "union_declarations": [],
  "type_alias_declarations": [],
  "declaration_order": [
    "test.name/SecondProtocol",
    "test.name/FirstProtocol",
    "test.name/SingleMemberService",
    "test.name/MultiHomogeneousMemberService",
    "test.name/MultiHeterogeneousMemberService"
  ],
  "declarations": {
    "test.name/FirstProtocol": "interface",
    "test.name/SecondProtocol": "interface",
    "test.name/SingleMemberService": "service",
    "test.name/MultiHeterogeneousMemberService": "service",
    "test.name/MultiHomogeneousMemberService": "service",
    "test.name/SomeLongAnonymousPrefix0": "struct",
    "test.name/SomeLongAnonymousPrefix1": "struct"
  }
}
